Services/ContainerReference/classes/ilContainerReferenceRepositorySelectorExplorerGUI.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+namespace ILIAS\ContainerReference;
+
+use ILIAS\Repository\StandardGUIRequest;
+use ilObjectDefinition;
+use ilObject;
+use ilAccessHandler;
+use ilTreeExplorerGUI;
+use ilContainerSorting;
+
+/**
+ * Explorer for selecting repository items.
+ *
+ * The implementation starts as a replacement for the often (ab)used ilSearchRootSelector class.
+ * Clicking items triggers a "selection" command.
+ * However ajax/checkbox/radio and use in an inputgui class should be implemented in the future, too.
+ *
+ * @author Alexander Killing <killing@leifos.de>
+ */
+class ilContainerReferenceRepositorySelectorExplorerGUI extends ilTreeExplorerGUI
```

Comment:
Furthermore it should be sufficient to use `extends ilRepositorySelectorExplorerGUI` and simply **overwrite** `isNodeClickable`